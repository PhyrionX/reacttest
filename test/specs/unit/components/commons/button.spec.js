import React from 'react';
import { render, fireEvent } from '@testing-library/react'
import Button from '../../../../../src/components/commons/button';

describe('<Button />', () => {
  it('should mount empty component', () => {
    const { asFragment } = render(<Button />);
    
    expect(asFragment()).toMatchSnapshot();
  })

  it('should mount component', () => {
    const mockClick = jest.fn();
    const { asFragment, getByLabelText } = render(<Button text="testText" onClick={ mockClick }/>);
    const button = getByLabelText('button');
    
    fireEvent.click(button);
    
    expect(asFragment()).toMatchSnapshot();
    expect(mockClick).toHaveBeenCalled();
  })

})