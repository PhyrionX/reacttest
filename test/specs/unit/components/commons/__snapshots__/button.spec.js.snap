// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should mount component 1`] = `
<DocumentFragment>
  <button
    aria-label="button"
    class="billin-button"
  >
    testText
  </button>
</DocumentFragment>
`;

exports[`<Button /> should mount empty component 1`] = `
<DocumentFragment>
  <button
    aria-label="button"
    class="billin-button"
  />
</DocumentFragment>
`;
